### ISSUE TITLE
[Improvement] Simplify compilation by compiling scripts

### ISSUE THREAD

**Issue by xunliu** [apache/gravitino#4266]
### Description
We will create a shell script (build.sh) that presents the user with a menu of build options. The user can select the desired build option, and the script will execute the corresponding build command. This reduces the need to manually type out build commands, making the process more convenient and less error-prone.

### Sample build.sh Script
```
#!/bin/bash  

# Define build options  
buildOptions=(  
    "Only compile and skip test:  ./gradlew clean build -x test"  
    "Only execute Unit test and skip Integration test:  ./gradlew test -PskipITs"  
    "Run the integration tests in embedded mode: ./gradlew build -x test && ./gradlew test -PskipTests -PtestMode=embedded"  
    "Run the integration tests in deploy mode: ./gradlew build -x test && ./gradlew compileDistribution && ./gradlew test -PskipTests -PtestMode=deploy"  
    "Exit"  
)  
```

I think we can refer some build command of 
1. https://github.com/apache/gravitino/tree/main/.github/workflows
2. https://github.com/apache/gravitino/blob/main/docs/how-to-build.md
3. https://github.com/apache/gravitino/blob/main/docs/how-to-test.md

```
# Display the selection menu  
echo "Please choose a build option:"  

PS3='Enter the number of your choice: '  # Set the PS3 prompt for select  

# Use select to create the menu  
select opt in "${buildOptions[@]}"  
do  
    case $opt in  
        "Build Option A: make buildA")  
            echo "You chose Build Option A"  
            make buildA  
            ;;  
        "Build Option B: make buildB")  
            echo "You chose Build Option B"  
            make buildB  
            ;;  
        "Build Option C: make buildC")  
            echo "You chose Build Option C"  
            make buildC  
            ;;  
        "Build Option D: make buildD")  
            echo "You chose Build Option D"  
            make buildD  
            ;;  
        "Exit")  
            echo "Exiting"  
            break  
            ;;  
        *)  
            echo "Invalid option, please choose again"  
            ;;  
    esac  
done  
```

### How to Use

1. Run the Script
Execute the script:

```
./build.sh  
```

### Explanation
+ Define Build Options:
The buildOptions array contains the list of build options, along with the commands that should be executed for each option.

+ Selection Menu:
The PS3 variable sets the prompt for the select command. The select command creates a loop with a menu that allows the user to choose an option by entering its corresponding number.

+ Case Statement:
The case statement executes the command associated with the selected option. For each option, the script executes the relevant build command.

+ Exit Option:
The Exit option allows the user to quit the script.

+ Example Build Commands
In this example, the build commands are make buildA, make buildB, make buildC, and make buildD. You can replace these commands with whatever build commands are relevant to your project.

This script significantly simplifies the process of executing different build commands and ensures that users only need to input the number corresponding to their desired task, reducing the potential for error and saving time.

**Comment by featherchen** [comment]
Hi, @xunliu. Since there are many build options can be set, e.g. those in `gradle.properties`, it might be complicated for users to set these properties each time executing `build.sh`. However, listing all the combinations of possible building options might be hard to understand and maintain.

Maybe we could record the previous building options somewhere (e.g. some JSON file) and ask the user whether to use it or modify it based on the script we write ?

**Comment by noidname01** [comment]
@featherchen I think we can still provide some most common used build option combinations (like the `buildOptions` listed above). 
If the user want advanced option, they just type on their own, we just provide a simplified, instinctive way for common situations.
WDYT?

**Comment by xunliu** [comment]
@mchades @yuqi1129 What do you think about this issue, Please give some suggestions, Thanks!

**Comment by featherchen** [comment]
> 1. It's a good idea to list some commonly used build options and combinations in the `kts` file.
> 
>     2. Putting all build options in a conspicuous place like in the `build.kts` will be preferable as advanced user may has custom needs.
> 
> 
> ```
> ### Define build options  
> commonlyUsedbuildOptions=(  
>     "Only compile and skip the unit test and Integration test:  ./gradlew clean build -x test"  
>     "Only execute the unit test and skip Integration test:  ./gradlew test -PskipITs"  
>     "Run the integration tests in embedded mode: ./gradlew build -x test && ./gradlew test -PskipTests -PtestMode=embedded"  
>     "Run the integration tests in deploy mode: ./gradlew build -x test && ./gradlew compileDistribution && ./gradlew test -PskipTests -PtestMode=deploy"  
>     "Exit"  
> )  
> 
> ### All build optional 
> 
> #### Test related
> - skipITs
> - skipDockerTests
> - testMode
> - jdbcBackend
> 
> #### Complile and assemble 
> 
> - gradle compileDistribution
> - gralde assemblyDistribution
> 
> ....
> ```

@yuqi1129 Thanks for the reply, so do you think we should offer all the combination of building options for the advance user to choose ? Won't it be too complicated to understand and does not match our goal of simplifying compilation ?

**Comment by yuqi1129** [comment]
> so do you think we should offer all the combination of building options for the advance user to choose?

No, it will be too tedious. Some of the most commonly used combinations will be sufficient. We can provide a list of options and let users combine them freely.
